### PR TITLE
PYTHON-2396 Deprecate ssl_keyfile and ssl_certfile URI options

### PR DESCRIPTION
**Pull request by prashantmital**
Should I create a new section in the PyMongo 4 migration guide for deprecated/remove URI options?

### CODE REVIEW

File: doc/examples/tls.rst [hunk 1]
```diff
@@ -118,13 +118,19 @@ PyMongo can be configured to present a client certificate using the
...
+``tlsCertificateKeyFile`` option::
 
   >>> client = pymongo.MongoClient('example.com',
   ...                              tls=True,
-  ...                              tlsCertificateKeyFile='/path/to/client.pem',
-  ...                              ssl_keyfile='/path/to/key.pem')
+  ...                              tlsCertificateKeyFile='/path/to/concat_client.pem')
 
 Python supports providing a password or passphrase to decrypt encrypted
 private keys. Use the ``tlsCertificateKeyFilePassword`` option::
```
Comment (ShaneHarvey) [inline doc/examples/tls.rst:136]: Hmmm, what's the upgrade path for users of this example? Can we remove `ssl_keyfile` here?

**Review by ShaneHarvey — CHANGES_REQUESTED** [review]
> Should I create a new section in the PyMongo 4 migration guide for deprecated/remove URI options?

Yes, you can add it under the MongoClient section.

Can you also replace Deprecate with Remove in the PR title?

File: doc/examples/tls.rst (hunk 1 above)
Comment (prashantmital) [reply doc/examples/tls.rst:136]: I am not sure I follow? Users will need to concat the files...

**Comment by prashantmital** [comment]
Ready for another look @ShaneHarvey. I have removed the migration notes as they will be added in https://jira.mongodb.org/browse/PYTHON-2162 
This change is intended to be merged into both v3.12 and master branches.

File: doc/examples/tls.rst [hunk 2]
```diff
@@ -118,24 +118,30 @@ PyMongo can be configured to present a client certificate using the
   ...                              tls=True,
   ...                              tlsCertificateKeyFile='/path/to/client.pem')
 
-If the private key for the client certificate is stored in a separate file use
-the ``ssl_keyfile`` option::
+If the private key for the client certificate is stored in a separate file,
+it should be concatenated with the certificate file. For example, to
+concatenate a PEM-formatted certificate file ``cert.pem`` and a PEM-formatted
+keyfile ``key.pem`` into a single file ``combined.pem``, on *nix systems users
```
Comment (ShaneHarvey) [inline doc/examples/tls.rst:124]: `*` means italics is RST. Does this render? Also everywhere else in the docs we just say "Unix" so maybe just change to that?

**Review by ShaneHarvey — CHANGES_REQUESTED** [review]
Please adjust the test suite to ignore deprecation warnings when using the old option names.

File: doc/examples/tls.rst (hunk 2 above)
Comment (prashantmital) [reply doc/examples/tls.rst:124]: Switched to Unix.

**Comment by prashantmital** [comment]
I've ignored deprecations in `test_ssl` but noticed that we were ignorning already deprecated options that were in that file like `ssl_cert_reqs` so I am not sure why we are doing so specifically for these newly deprecated options? 
Is the intention to ignore the deprecations in all other test modules other than SSL?

**Review by ShaneHarvey — APPROVED** [review]
Not sure I fully understand your question. In general we should use `ignore_deprecations` whenever we're testing a deprecating feature otherwise the test output is cluttered with DeprecationWarnings.

**Pull request merged**